### PR TITLE
Move to Python 3

### PR DESCRIPTION
**Pull request by fbertsch**
This is in preparation for the Glean support

### CODE REVIEW

**Comment by fbertsch** [comment]
Ran this locally, and was fine (for versions 64+).

Testing the job on a cluster where it will run in production.

**Comment by fbertsch** [comment]
Following the fix in https://github.com/travis-ci/travis-ci/issues/9815.

File: probe_scraper/parsers/third_party/parse_events.py [hunk 1]
```diff
@@ -53,89 +53,89 @@ def check(self, identifier, key, value):
...
         changed_fields = OPTIONAL_FIELDS
-    changed_fields['record_in_processes'] = TypeChecker(list, basestring)
+    changed_fields['record_in_processes'] = TypeChecker(list, str)
 
     ALL_FIELDS = REQUIRED_FIELDS.copy()
     ALL_FIELDS.update(OPTIONAL_FIELDS)
 
     # Check that all the required fields are available.
-    missing_fields = [f for f in REQUIRED_FIELDS.keys() if f not in definition]
+    missing_fields = [f for f in list(REQUIRED_FIELDS.keys()) if f not in definition]
```
Comment (mdboom) [inline probe_scraper/parsers/third_party/parse_events.py]: This `list` is probably not strictly necessary, but it's a minor point.

File: probe_scraper/parsers/third_party/parse_events.py [hunk 2]
```diff
@@ -53,89 +53,89 @@ def check(self, identifier, key, value):
...
     if len(missing_fields) > 0:
         raise KeyError(identifier + ' - missing required fields: ' + ', '.join(missing_fields))
 
     # Are there any unknown field?
     if strict:
-        unknown_fields = [f for f in definition.keys() if f not in ALL_FIELDS]
+        unknown_fields = [f for f in list(definition.keys()) if f not in ALL_FIELDS]
```
Comment (mdboom) [inline probe_scraper/parsers/third_party/parse_events.py]: Ditto about `list`.

File: probe_scraper/parsers/third_party/usecounters.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 import collections
 import re
-import StringIO
+import io
```
Comment (mdboom) [inline probe_scraper/parsers/third_party/usecounters.py]: It looks like maybe this is an unused import to begin with?

File: probe_scraper/runner.py [hunk 4]
```diff
@@ -147,6 +147,8 @@ def check_git_probe_structure(data):
 def load_git_probes(cache_dir, out_dir, repositories_file, dry_run):
     repositories = RepositoriesParser().parse(repositories_file)
     commit_timestamps, repos_probes_data, emails = git_scraper.scrape(cache_dir, repositories)
+    print("Scraped")
+    print(commit_timestamps)
```
Comment (mdboom) [inline probe_scraper/runner.py]: Did you intend to add stuff here?

File: probe_scraper/scrapers/git_scraper.py [hunk 5]
```diff
@@ -33,20 +33,29 @@ def retrieve_files(repo_info, cache_dir):
...
+                        f.write(contents.encode("UTF-8"))
+                    print("Hash: " + str(_hash))
+                    print("ptype: " + str(ptype))
+                    print("disk path: " + str(disk_path))
+                    
                 results[_hash][ptype].append(disk_path)
+                print(results[_hash][ptype])
                 timestamps[_hash] = ts
     except Exception:
+        print("Got an exception!")
```
Comment (mdboom) [inline probe_scraper/scrapers/git_scraper.py]: Are these new prints for keeps?

File: requirements.txt [hunk 6]
```diff
@@ -6,3 +6,4 @@ pyyaml
 requests
 requests_cache
 schema
+urllib3==1.23
```
Comment (mdboom) [inline requirements.txt]: Where were the uses of this added?

File: probe_scraper/runner.py (hunk 4 above)
Comment (fbertsch) [reply probe_scraper/runner.py]: Nope!

File: probe_scraper/scrapers/git_scraper.py (hunk 5 above)
Comment (fbertsch) [reply probe_scraper/scrapers/git_scraper.py]: Wow, could have sworn I removed them before committing. Thanks!

File: requirements.txt (hunk 6 above)
Comment (fbertsch) [reply requirements.txt]: Added a comment. This has to do with the requests_cache library and `ordered_dict`.

File: requirements.txt [hunk 7]
```diff
@@ -6,3 +6,4 @@ pyyaml
 requests
 requests_cache
 schema
+urllib3==1.23 # See https://github.com/urllib3/urllib3/issues/1456#issuecomment-430404098
```
Comment (mdboom) [inline requirements.txt]: Fun! :)

**Pull request merged**